doc/test_organization/managing_tests_dependencies.qbk

```diff
@@ -7,17 +7,31 @@
 
 [section:tests_dependencies Managing test dependencies]
 
-Dependencies among tests are managed by means of [link boost_test.tests_organization.decorators decorators]. 
+In general, you should make sure that test cases are independent on one another.
```

Comment:
My view is that all of those orders are implementation details. If some test should be run after another one, then the user should use depends_on decorator. If the suite level entry/exit fixture should be detailed (in terms of number of execution of the fixture itself), I think this is enough to say that the test tree is processed in a breadth first manner, which is exactly that. When I shuffle the test, I am expecting that there is no order at all, and test cases are picked randomly. If a test case is to be executed, and if I have random, I am (as a user) expecting that the execution is postponed until all parent test cases are executed (which is not the same as ordering by rank). So the algorithm should look like this:

    while not queue.empty():
          current_test_case = queue.pop()
          if not current_test_case.all_parents_executed():
              queue.push_back(current_test_case)
              continue
          
          # execute test 

Here, with random, I am expecting that queue has been shuffled.